Indicators/TomDemarkSequential.cs

```diff
@@ -0,0 +1,353 @@
+                _currentPhase = TomDemarkSequentialPhase.SellCountdown;
+                _supportPrice = window.Skip(window.Count - MaxSetupCount).Take(MaxSetupCount).Min(b => b.Low);
+                _stepCount = 0;
+
+                return EncodeState(
+                    isPerfect ? TomDemarkSequentialPhase.SellSetupPerfect : TomDemarkSequentialPhase.SellSetup, MaxSetupCount);
+            }
+
+            return EncodeState(TomDemarkSequentialPhase.SellSetup, _stepCount);
+        }
+        
+        ResetStepAndPhase();
+        return Default;
+    }
+
+    private decimal HandleBuySetupPhase(
+        IBaseDataBar current,
+        IBaseDataBar bar4Ago,
+        IReadOnlyWindow<IBaseDataBar> window
+        )
+    {
+        if (current.Close < bar4Ago.Close)
+        {
+            _stepCount++;
+            if (_stepCount == MaxSetupCount)
+            {
+                var isPerfect = IsBuySetupPerfect(window);
+                _currentPhase = TomDemarkSequentialPhase.BuyCountdown;
+                _resistancePrice = window.Skip(window.Count - MaxSetupCount).Take(MaxSetupCount).Max(b => b.High);
+                _stepCount = 0;
+
+                return EncodeState(
+                    isPerfect ? TomDemarkSequentialPhase.BuySetupPerfect : TomDemarkSequentialPhase.BuySetup, MaxSetupCount);
+            }
+
+            return EncodeState(TomDemarkSequentialPhase.BuySetup, _stepCount);
+        }
+
+        ResetStepAndPhase();
+        return Default;
+    }
+
+    private decimal InitializeSetupPhase(IBaseDataBar current, IBaseDataBar bar4Ago)
+    {
+        if (current.Close < bar4Ago.Close)
+        {
+            _currentPhase = TomDemarkSequentialPhase.BuySetup;
+            _stepCount = 1;
+
+            return EncodeState(TomDemarkSequentialPhase.BuySetup, _stepCount);
+        }
+
+        if (current.Close > bar4Ago.Close)
+        {
+            _currentPhase = TomDemarkSequentialPhase.SellSetup;
+            _stepCount = 1;
+
+            return EncodeState(TomDemarkSequentialPhase.SellSetup, _stepCount);
+        }
+
+        return Default;
+    }
+
+    private void ResetStepAndPhase()
+    {
+        _stepCount = 0;
+        _currentPhase = TomDemarkSequentialPhase.None;
+    }
+    private static bool IsBuySetupPerfect(IReadOnlyWindow<IBaseDataBar> window)
+    {
+        var bar6 = window[3];
+        var bar7 = window[2];
+        var bar8 = window[1];
+        var bar9 = window[0];
+
+        return bar8.Low <= bar6.Low && bar8.Low <= bar7.Low ||
+            bar9.Low <= bar6.Low && bar9.Low <= bar7.Low;
+    }
+
+    private static bool IsSellSetupPerfect(IReadOnlyWindow<IBaseDataBar> window)
+    {
+        var bar6 = window[3];
+        var bar7 = window[2];
+        var bar8 = window[1];
+        var bar9 = window[0];
+
+        return bar8.High >= bar6.High && bar8.High >= bar7.High ||
+            bar9.High >= bar6.High && bar9.High >= bar7.High;
```

Comment:
Should these be just `>`/`<`  not sure the `=` is valid in the definition? ref https://practicaltechnicalanalysis.blogspot.com/2013/01/tom-demark-sequential.html